prototype2/efu/Loader.cpp

```diff
@@ -25,6 +24,17 @@ void Loader::unloadPlugin() {
 }
 
 bool Loader::loadPlugin(const std::string lib) {
+  try {
+    auto &FoundModule = DetectorModuleRegistration::find(lib);
+    ParserPopulator = FoundModule.CLISetup;
+    myFactory = FoundModule.DetectorFactory.get();
+    XTRACE(INIT, INF, "Loaded statically linked detector module.");
+    return true;
+  } catch (std::runtime_error &Error) {
+    XTRACE(INIT, INF, "Unable to find statically linked detector module with "
+                      "name\"%s\". Attempting to open external plugin.",
+           lib.c_str());
+  }
   std::string libname = "./" + lib + ".so";
```

Comment:
I have always wanted to get rid of this allowing both "plugin" and "plugin.so" - since you're messing with this code, could I persuade you to add the logic ?

Reply:
Sure. I fix.